### ISSUE TITLE
NPE in vertx:package when using vertx-camel-bridge

### ISSUE THREAD

**Issue by davsclaus** [fabric8io/vertx-maven-plugin#64]
I get this error
```
[ERROR] Failed to execute goal io.fabric8:vertx-maven-plugin:1.0.1:package (default-cli) on project chapter7-vertx-camel: Unable to build fat jar: Unable to combine SPI files for chapter7-vertx-camel: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:vertx-maven-plugin:1.0.1:package (default-cli) on project chapter7-vertx-camel: Unable to build fat jar
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoFailureException: Unable to build fat jar
	at io.fabric8.vertx.maven.plugin.mojos.PackageMojo.execute(PackageMojo.java:144)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: org.apache.maven.plugin.MojoExecutionException: Unable to combine SPI files for chapter7-vertx-camel
	at io.fabric8.vertx.maven.plugin.utils.PackageHelper.combineServiceProviders(PackageHelper.java:248)
	at io.fabric8.vertx.maven.plugin.mojos.PackageMojo.execute(PackageMojo.java:129)
	... 21 more
Caused by: java.lang.NullPointerException
	at io.fabric8.vertx.maven.plugin.utils.ServiceCombinerUtil.findSPIsFromDependencies(ServiceCombinerUtil.java:149)
	at io.fabric8.vertx.maven.plugin.utils.ServiceCombinerUtil.combine(ServiceCombinerUtil.java:75)
	at io.fabric8.vertx.maven.plugin.utils.PackageHelper.combineServiceProviders(PackageHelper.java:218)
	... 22 more
```

When I am packaging the application using vertx:package (or doing a mvn install).
The only thing I have done is to add a maven dependency on vertx-camel-bridge. If I remove that dependency then the verx plugin can build the far jar

The example is located at
https://github.com/camelinaction/camelinaction2/tree/master/chapter7/vertx-camel

**Comment by kameshsampath** [comment]
@davsclaus can you try adding this configuration to v-m-p, and let me know if you are getting the same NPE ?

```
<configuration>
  <serviceProviderCombination>none</serviceProviderCombination>
</configuration>
```

**Comment by kameshsampath** [comment]
WIP

**Comment by davsclaus** [comment]
Yes now it works with that none configuration

**Comment by kameshsampath** [comment]
@davsclaus what I found with my debug is that when Shrinkwrap reads a `Asset` from a Archive path, and lets assume its a directory e.g `META-INF/services/org`, then we can't open the stream on it because its directory and hence Shrinkwrap will return null Asset for it. I have now put null check there to make sure we don't read `null` assets

**Comment by kameshsampath** [comment]
this is a temp workaround @davsclaus , https://github.com/fabric8io/vertx-maven-plugin/pull/66 should have the permanent fix once its merged, and you dont need to add the config change as said above

**Comment by kameshsampath** [comment]
resolved via #66 
@davsclaus, please verify before we can close this

**Comment by davsclaus** [comment]
Yes it works with SNAPSHOT
```
[INFO] --- vertx-maven-plugin:1.0-SNAPSHOT:package (vmp) @ chapter7-vertx-camel ---
[INFO]
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ chapter7-vertx-camel ---
[INFO] Installing /Users/davsclaus/workspace/camelinaction2/chapter7/vertx-camel/target/livescore.jar to /Users/davsclaus/.m2/repository/com/camelinaction/chapter7-vertx-camel/2.0.0/chapter7-vertx-camel-2.0.0.jar
[INFO] Installing /Users/davsclaus/workspace/camelinaction2/chapter7/vertx-camel/pom.xml to /Users/davsclaus/.m2/repository/com/camelinaction/chapter7-vertx-camel/2.0.0/chapter7-vertx-camel-2.0.0.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

**Comment by kameshsampath** [comment]
closed via #66
